adafruit_requests.py

```diff
@@ -762,17 +789,33 @@ def wrap_socket(
         return _FakeSSLSocket(socket, self._iface.TLS_MODE)
 
 
+def create_fake_ssl_context(
+    pool: SocketpoolModuleType, iface: Optional[InterfaceType] = None
+) -> _FakeSSLContext:
+    """Legacy API for creating a fake SSL context"""
+    if not iface:
+        # pylint: disable=protected-access
+        iface = pool._the_interface
+    pool.set_interface(iface)
+    return _FakeSSLContext(iface)
+
+
+def set_socketpool(
```

Comment:
In my opinion `set_socket` isn't the right name, since we are actually setting a fake `Socketpool`

Reply:
Also, with `create_fake_ssl_context` you can do:
```
ssl_context = adafruit_requests.create_fake_ssl_context(pool, esp)
requests = adafruit_requests.Session(pool, ssl_context, socket_handler)
```
_with or without adding `socket_handler`_

So could completely remove `set_socket` and the new `set_socketpool`